static/js/tools/choropleth/choropleth_template.tsx

```diff
@@ -14,28 +14,122 @@
  * limitations under the License.
  */
 
-import React, { Component } from "react";
+/** React component for the entire choropleth page with sidebar and map. */
 
-class MainPane extends Component {
-    constructor(props) {
-      super(props);
-    }
-  
-    render() {
-        //TODO(iancostello): remove bounding-box and centroid if unneccesary.
-        //TODO(iancostello):Add dynamic sizing.
-        return ( 
-            <React.Fragment>
-                <div className="column" id="breadcrumbs"></div>
-                <svg id="map_container" width="800px" height="500px">
-                    <g className="map"></g>
-                    <g className="bounding-box"><rect></rect></g>
-                    <g className="centroid"><circle r="4"></circle></g>
-                </svg>
-                <div className="column" id="hover-text-display"></div>
-            </React.Fragment>
-        );
+import React from "react";
+import { Menu } from "../statsvar_menu";
+const axios = require("axios");
+import {
+  PageStateType
+} from "../timeline_page"
+
+class MainPane extends React.Component<{}, PageStateType> {
+  constructor(props) {
+    super(props);
+    // TODO(iancostello): Handle bad inputs.
+
+    // Add function bindings.
+    this._togglePerCapita = this._togglePerCapita.bind(this);
+
+    // Get default values for optional fields.
+    var urlParams = new URLSearchParams(window.location.search);
+    var isPerCapita = false
+    if (urlParams.has("perCapita")) {
+      isPerCapita = ["true", "t", "1"].includes(
+        urlParams.get("perCapita").toLowerCase());
     }
+    this.state = {
+      statsVarPaths: [],
+      statsVarInfo: {},
+      perCapita: isPerCapita,
+      places: [],
+      statsVarValid: new Set(),
```

Comment:
May we have comments on each of these fields?  This is the main data structure I believe.